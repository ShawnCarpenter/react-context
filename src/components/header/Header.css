.Header {
  display:flex;
  justify-content: space-around;
  align-items: center;
  min-width:100vw;

  button:hover {
    cursor:pointer;
  }
}
