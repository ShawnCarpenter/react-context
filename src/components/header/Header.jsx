import React from 'react';
import { useSwitchTheme } from '../../hooks/theme';

const Header = () => {
  const switchTheme = useSwitchTheme();
 
  return (
    <div>
      <h1>Header</h1>
      <button onClick={switchTheme}>Hit the lights!</button>
    </div>
  );
};

export default Header;
