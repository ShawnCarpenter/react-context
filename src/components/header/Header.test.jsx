import React from 'react';
import { render } from '@testing-library/react';
import Header from './Header';

jest.mock('../../hooks/theme.js');

describe('Header component', () => {
  it('renders Header', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
