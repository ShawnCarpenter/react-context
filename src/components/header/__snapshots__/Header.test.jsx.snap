// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header 1`] = `
<DocumentFragment>
  <div
    class="Header"
  >
    <h1>
      Themes, yay...
    </h1>
    <button>
      Hit the lights!
    </button>
  </div>
</DocumentFragment>
`;
